llpc/context/llpcCompiler.cpp

```diff
@@ -506,11 +506,6 @@ Result Compiler::BuildShaderModule(const ShaderModuleBuildInfo *shaderInfo, Shad
     result = Result::ErrorInvalidShader;
 
   if (moduleDataEx.common.binType == BinaryType::Spirv) {
-    // Dump SPIRV binary
```

Comment:
I don't think we should move the dump logic.
- some app only call build shader module without call create pipeline, move this logic will causes some spirv binary isn't dumpped.
- if we enable shader module compile during build shader module, the binary type in shaderInfo->pModuleData is LLVM IR instead of SPIRV binary, it causes we can't SPIRV binary correctly.

Have you tried to call createDirectory in  PipelineDumper::DumpSpirvBinary? I hope it can solve your issue.

Reply:
I can change this to createDirectory in PipelineDumper::DumpSpirvBinary, but I wasn't sure if that made sense design-wise as the other use of createDirectory was within a mutex.  Although I see no practical issue with using createDirectory.

Reply:
we only need mutex for new file,  I think it is safe to call createDirectory without mutex.